<launch>
	<!-- FROBIT PARAMETERS --> 
	<param name="diff_steer_wheel_distance" value="0.375"/>
	<param name="ticks_per_meter_left" value="677" />
	<param name="ticks_per_meter_right" value="677" />

	<group ns="fmDecisionMakers">
		<node pkg="frobit_demo" name="mission" type="keyboard_mission_node.py" output="screen">
			<param name="keyboard_sub" value="/fmHMI/keyboard" />
			<param name="automode_pub" value="/fmPlan/automode" />
    		<param name="deadman_pub" value="/fmSafe/deadman" />
			<param name="cmd_vel_pub" value="/fmCommand/cmd_vel" />
            <param name="max_linear_velocity" value="1.0"/>
            <param name="max_angular_velocity" value="0.8"/>
            <param name="linear_velocity_step" value="0.1"/>
            <param name="angular_velocity_step" value="0.2"/>
		</node>
	</group>

	<node pkg="keyboard_interface" type="keyboard_node" name="keyboard">
		<param name="keyboard_pub" value="/fmHMI/keyboard" />
	</node>

	<node pkg="twist_stamped_to_twist" name="converter" type="cmd_vel_converter.py">	
		<param name="cmd_vel_sub" value="/fmCommand/cmd_vel" />
		<param name="cmd_vel_pub" value="/cmd_vel" />
	</node>


	<!-- Notice: Use pkg="stage" if running ROS Groovy and pkg="stage_ros" if running ROS Hydro --> 
 	<node pkg="stage_ros" type="stageros" name="simulator" args="/home/martin/roswork/src/frobo_rsd/Navigation_Setup/maps/boxSimTest.world"/>
	<remap from="/fmKnowledge/pose" to="/odom" />

	<node pkg="frobit_plot" name="frobit_plot" type="frobit_plot_node.py" output="screen" >
			<param name="pose_sub" value="/fmKnowledge/pose" />
			<param name="gnss_sub" value="/fmInformation/gpgga_tranmerc" />
			<param name="odom_sub" value="/fmKnowledge/odom" />
			<param name="wptnav_status_sub" value="/fmInformation/wptnav_status" />
			<param name="plot_pose_track" value="true" />
			<param name="plot_gnss_track" value="false" />
			<param name="plot_odometry_track" value="false" />
			<param name="plot_pose_yaw" value="false" />
			<param name="map_trackpoint_threshold" value="0.02" />
			<param name="map_max_trackpoints" value="5000" />
			<param name="map_minimum_size" value="2.0" />
			<param name="map_easting_offset" value="0.0" />
			<param name="map_northing_offset" value="0.0" />
			<param name="map_update_frequency" value="10.0" />
			<param name="map_title" value="Frobit track" />
			<param name="map_window_size" value="6.5" />
			<param name="avatar_extension_front" value="0.29" />
			<param name="avatar_extension_rear" value="0.06" />
			<param name="avatar_extension_lateral" value="0.13" />
			<param name="avatar_reverse" value="false" />
		</node>

	<node pkg="move_base" name="move_base" type="move_base" >
		<param name="footprint_padding" value="0.01" />
		<param name="controller_frequency" value="10.0" />
		<param name="controller_patience" value="3.0" />
		
		<param name="oscillation_timeout" value="30.0" />
		<param name="oscillation_distance" value="0.5" />

		<rosparam file="../Navigation_Setup/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap"/>
		<rosparam file="../Navigation_Setup/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap"/>
		<rosparam file="../Navigation_Setup/move_base_config/local_costmap_params.yaml" command="load"/>
		<rosparam file="../Navigation_Setup/move_base_config/global_costmap_params.yaml" command="load"/>
		<rosparam file="../Navigation_Setup/move_base_config/base_local_planner_params.yaml" command="load"/>
	</node>


  	<node name="rviz" pkg="rviz" type="rviz" args="-d move_base_config/move_base_config.rviz" />

	<!--node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 map odom 50"  >
	</node-->

	<node name="map_server" pkg="map_server" type="map_server" args="/home/martin/roswork/src/frobo_rsd/Navigation_Setup/maps/map2.yaml" />

	<node pkg="amcl" type="amcl" name="amcl" respawn="true">
	  <remap from="scan" to="base_scan" />
	
	  <param name="odom_model_type" value="omni"/>
	  <param name="odom_alpha5" value="0.1"/>
	  <param name="transform_tolerance" value="0.2" />
	  <param name="gui_publish_rate" value="10.0"/>
	  <param name="laser_max_beams" value="30"/>
	  <param name="laser_min_range" value="0.1"/>
	  <param name="min_particles" value="500"/>
	  <param name="max_particles" value="5000"/>
	  <param name="kld_err" value="0.05"/>
	  <param name="kld_z" value="0.99"/>
	  <param name="odom_alpha1" value="0.2"/>
	  <param name="odom_alpha2" value="0.2"/>
	
	  <param name="odom_alpha3" value="0.8"/>
	  <param name="odom_alpha4" value="0.2"/>
	  <param name="laser_z_hit" value="0.5"/>
	  <param name="laser_z_short" value="0.05"/>
	  <param name="laser_z_max" value="0.05"/>
	  <param name="laser_z_rand" value="0.5"/>
	  <param name="laser_sigma_hit" value="0.2"/>
	  <param name="laser_lambda_short" value="0.1"/>
	  <param name="laser_lambda_short" value="0.1"/>
	  <param name="laser_model_type" value="likelihood_field"/>
	
	  <param name="laser_likelihood_max_dist" value="2.0"/>
	  <param name="update_min_d" value="0.2"/>
	  <param name="update_min_a" value="0.5"/>
	  <param name="odom_frame_id" value="odom"/>
	  <param name="odom_frame_id" value="odom"/>
	  <param name="global_frame_id" value="map"/>

	  <param name="resample_interval" value="1"/>
	  <param name="transform_tolerance" value="0.1"/>
	  <param name="recovery_alpha_slow" value="0.0"/>
	  <param name="recovery_alpha_fast" value="0.0"/>
	<param name="initial_pose_x" value="10"/>
	<param name="initial_pose_y" value="5"/>
	<param name="initial_cov_xx" value="50"/>
	<param name="initial_cov_yy" value="50"/>
	<param name="initial_cov_aa" value="6"/>


	 
	</node>



</launch>

